File: Skydown App/Models/User/ProfileGalleryItem.swift
import Foundation

enum ProfileMediaType: String, CaseIterable, Identifiable, Codable {
    case image

    var id: String { rawValue }

    var title: String {
        "Bilder"
    }

    var systemImage: String {
        "photo.fill"
    }

    var storageFolder: String {
        "gallery"
    }
}

struct ProfileGalleryItem: Codable, Identifiable {
    var id: String?
    var ownerId: String
    var type: String
    var title: String
    var caption: String?
    var mediaURL: String
    var thumbnailURL: String?
    var createdAt: Date

    var mediaType: ProfileMediaType {
        ProfileMediaType(rawValue: type) ?? .image
    }
}

File: androidApp/src/main/java/com/skydown/android/ui/model/ProfileGalleryItem.kt
package com.skydown.android.ui.model

data class ProfileGalleryItem(
    val id: String,
    val ownerId: String,
    val type: ProfileMediaType,
    val title: String,
    val caption: String?,
    val mediaUrl: String,
    val thumbnailUrl: String?,
    val storagePath: String?,
    val createdAtEpochMillis: Long,
)

enum class ProfileMediaType(
    val rawValue: String,
    val title: String,
    val fallbackExtension: String,
) {
    Image("image", "Bilder", "jpg");

    companion object {
        fun fromRawValue(rawValue: String?): ProfileMediaType = Image
    }
}
